### PR TITLE
install: add directory permissions check before building binaries

### LINKED ISSUE #413: Update logic of tt install with system cfg
Now on macOS, the `tt install tarantool` command with the system config, done without `sudo`, builds tarantool and failed when copying the binaries into `/opt/tarantool/bin` because don't have an access rights.
One or both must be done:
- Check for access to install the binary before building  the tarantool
- offer to enter the password and with updated access rights copy the binaries after a successful build

### PR DESCRIPTION
**Pull request by vr009**
Closes #413

### CODE REVIEW

File: cli/install/install.go [hunk 1]
```diff
@@ -899,6 +899,19 @@ func installTarantool(binDir string, incDir string, installCtx InstallCtx,
 	return nil
 }
 
+func dirsAreWriteable(dirs []string) bool {
+	for _, dir := range dirs {
+		tempFileName := filepath.Join(dir, "tempfile")
+		err := os.WriteFile(tempFileName, []byte{}, 0)
```
Comment (LeonidVas) [inline cli/install/install.go:905]: Why don't you use `os.Stat()`?

File: cli/install/install.go [hunk 2]
```diff
@@ -1061,6 +1074,13 @@ func Install(binDir string, includeDir string, installCtx InstallCtx,
 	local string, cliOpts *config.CliOpts) error {
 	var err error
 
+	// This check is needed for knowing that we will be able to copy
+	// recently built binaries to the corresponding bin and include directories.
+	if !dirsAreWriteable([]string{binDir, includeDir}) {
+		return fmt.Errorf("some of directories are not writeable for the current user. " +
+			"Please, use sudo for the successful install")
```
Comment (LeonidVas) [inline cli/install/install.go:1081]: You don't have to use `sudo`, you can just update rights (the user have a choice)?

File: cli/install/install.go (hunk 1 above)
Comment (vr009) [reply cli/install/install.go]: I use the unix syscall `access` for finding out if the current user has write access.

File: cli/install/install.go [hunk 3]
```diff
@@ -1061,6 +1074,13 @@ func Install(binDir string, includeDir string, installCtx InstallCtx,
 	local string, cliOpts *config.CliOpts) error {
 	var err error
 
+	// This check is needed for knowing that we will be able to copy
+	// recently built binaries to the corresponding bin and include directories.
+	if !dirsAreWriteable([]string{binDir, includeDir}) {
+		return fmt.Errorf("some of directories are not writeable for the current user. " +
```
[the remark this answers is not in the document]
Comment (vr009) [reply cli/install/install.go]: Fixed.

File: cli/install/install.go (hunk 2 above)
Comment (vr009) [reply cli/install/install.go]: Fixed the output, thank you.

**Review by LeonidVas — APPROVED** [review]

**Pull request merged**